src/background/services/connectedAccountService.ts

```diff
@@ -0,0 +1,326 @@
+              name: caName,
+              origin: caOrigin2 ? caOrigin1 + '/' + caOrigin2 : caOrigin1,
+              img: makeBlockie(caName),
+              accountActive: ca.status.isMain,
+            },
+            statusName: Status.Connected,
+            statusMessage: 'Connected',
+            closeness: i + 1,
+          })
+        })
+      )
+    }
+
+    // *** REJECTED ***
+    if (prevPairs) {
+      const prevPendingPairs = prevPairs.filter(
+        (pair) => pair.statusName && pair.statusName === Status.Processing
+      )
+      const resolvedPairs = prevPendingPairs.filter(
+        (prevPair) => !newPendingIds.includes(prevPair.pendingRequestId!)
+      )
+      if (resolvedPairs.length !== 0) {
+        for (const resolvedPair of resolvedPairs) {
+          const requestStatus: 'not found' | 'pending' | 'approved' | 'rejected' =
+            await this.getRequestStatus(resolvedPair.pendingRequestId!)
+          if (requestStatus !== 'rejected') continue
+
+          const newPairsLengthBeforeFilter = newPairs.length
+          newPairs = newPairs.filter(
+            (p) =>
+              !(
+                p.firstAccount.name === resolvedPair.firstAccount.name &&
+                p.firstAccount.origin === resolvedPair.firstAccount.origin &&
+                p.secondAccount.name === resolvedPair.secondAccount.name &&
+                p.secondAccount.origin === resolvedPair.secondAccount.origin
+              ) &&
+              !(
+                p.secondAccount.name === resolvedPair.firstAccount.name &&
+                p.secondAccount.origin === resolvedPair.firstAccount.origin &&
+                p.firstAccount.name === resolvedPair.secondAccount.name &&
+                p.firstAccount.origin === resolvedPair.secondAccount.origin
+              )
+          )
+          newPairs.unshift({
+            firstAccount: resolvedPair.firstAccount,
+            secondAccount: resolvedPair.secondAccount,
+            statusName: Status.Error,
+            statusMessage:
+              newPairsLengthBeforeFilter === newPairs.length
+                ? 'Connection rejected'
+                : 'Disconnection rejected',
+            closeness: 1,
+          })
+        }
+      }
+    }
+    return newPairs
+  }
+
+  // ***** CALL *****
+
+  public async requestVerification(
+    props: {
+      firstAccountId: string
+      firstOriginId: string
+      secondAccountId: string
+      secondOriginId: string
+      isUnlink: boolean
+      firstProofUrl?: string
+      secondProofUrl?: string
+    },
+    stake: number
+  ): Promise<number> {
+    const {
+      firstAccountId,
+      firstOriginId,
+      secondAccountId,
+      secondOriginId,
+      isUnlink,
+      firstProofUrl,
+      secondProofUrl,
+    } = props
+    const contract = await this._getContract()
+    return contract.requestVerification(
+      {
+        firstAccountId,
+        firstOriginId,
+        secondAccountId,
+        secondOriginId,
```

Comment:
Is it really important to separate `origin` and `account`? Does the contract process these values?

Reply:
They are used in several messages:
https://github.com/dapplets/connected-accounts-assembly/blob/7a54cfd4cc912462c6511328bfc42e0a4cba9f0d/assembly/index.ts#L472
https://github.com/dapplets/connected-accounts-assembly/blob/7a54cfd4cc912462c6511328bfc42e0a4cba9f0d/assembly/index.ts#L477
https://github.com/dapplets/connected-accounts-assembly/blob/7a54cfd4cc912462c6511328bfc42e0a4cba9f0d/assembly/index.ts#L482
etc

Reply:
ok!